Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -11,10 +11,157 @@
 
 using namespace AZ;
 
-AllocatorBase::AllocatorBase(IAllocatorAllocate* allocationSource, const char* name, const char* desc) :
-    IAllocator(allocationSource),
-    m_name(name),
-    m_desc(desc)
+#define RECORDING_ENABLED 0
+
+#if RECORDING_ENABLED
+
+#include <AzCore/std/containers/map.h>
+#include <AzCore/IO/SystemFile.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/parallel/scoped_lock.h>
+
+namespace
+{
+    class DebugAllocator
+    {
+    public:
+        typedef void* pointer_type;
+        typedef AZStd::size_t size_type;
+        typedef AZStd::ptrdiff_t difference_type;
+        typedef AZStd::false_type allow_memory_leaks; ///< Regular allocators should not leak.
```

Comment:
We should use `using` here over typedef

```suggestion
        using pointer_type =  void*;
        using size_type = AZStd::size_t;
        using difference_type = AZStd::ptrdiff_t;
        using allow_memory_leaks = AZStd::false_type; ///< Regular allocators should not leak.
```